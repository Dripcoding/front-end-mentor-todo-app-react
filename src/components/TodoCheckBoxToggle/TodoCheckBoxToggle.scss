@import "../../palette.scss";

.todoCheckBoxToggle__container {
  border: 1px solid $very-dark-grayish-blue-one;
  border-radius: 50%;
  margin-right: 24px;
  height: 24px;
  width: 24px;
}