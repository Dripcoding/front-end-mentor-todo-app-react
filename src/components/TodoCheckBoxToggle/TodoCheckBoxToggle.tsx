import React from "react";

import "./TodoCheckBoxToggle.scss";

interface ITodoCheckBoxToggle {
  disabled: boolean;
}

const STYLE_BASE: string = "todoCheckBoxToggle__";

const TodoCheckBoxToggle = ({ disabled = true }: ITodoCheckBoxToggle): JSX.Element => {
  return (
    <div
      className={`${STYLE_BASE}container`}
      data-testid={"TODO_CHECKBOX_TOGGLE"}
    >
    </div>
  );
};

export default TodoCheckBoxToggle;