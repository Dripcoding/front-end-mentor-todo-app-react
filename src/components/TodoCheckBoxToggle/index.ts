import TodoCheckBoxToggle from "./TodoCheckBoxToggle";

export default TodoCheckBoxToggle;