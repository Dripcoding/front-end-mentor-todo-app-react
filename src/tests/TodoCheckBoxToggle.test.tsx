import { render } from "@testing-library/react";
import TodoCheckBoxToggle from "../components/TodoCheckBoxToggle";

describe("TodoCheckBoxToggle", () => {
  it("matches snapshot", () => {
    const { container } = render(<TodoCheckBoxToggle />);
    expect(container).toMatchSnapshot();
  });

  it("render correctly", () => {
    const { getByTestId } = render(<TodoCheckBoxToggle />);

    const container = getByTestId("TODO_CHECKBOX_TOGGLE");
    expect(container).toBeInTheDocument();
    expect(container).toBeVisible();
  });
});