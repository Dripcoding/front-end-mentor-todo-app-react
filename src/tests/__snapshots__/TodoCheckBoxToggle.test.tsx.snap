// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoCheckBoxToggle matches snapshot 1`] = `
<div>
  <div
    class="todoCheckBoxToggle__container"
    data-testid="TODO_CHECKBOX_TOGGLE"
  />
</div>
`;
